pages/index.js

```diff
@@ -2,34 +2,76 @@ import styled from "styled-components";
 import { EntertainmentIcon } from "@/components/Icons";
 import { AppliancesIcon } from "@/components/Icons";
 import { WorkIcon } from "@/components/Icons";
+import { useState } from "react";
+import { useEffect } from "react";
+
 const CATEGORY_MAP = {
   Entertainment: <EntertainmentIcon />,
   Appliances: <AppliancesIcon />,
   Work: <WorkIcon />,
 };
 
 export default function Home({ devices }) {
+  const [showDetails, setShowDetails] = useState(devices);
+
+  function handleButtonClick(id) {
+    setShowDetails(
+      showDetails.map((item) => {
+        if (item.id === id) {
+          return { ...item, isActive: !item.isActive };
+        } else {
+          return item;
+        }
+      })
+    );
+  }
+  useEffect(() => {
+    function handleDevices() {
+      const temp = showDetails.map((item) => {
+        return { ...item, isActive: true };
+      });
+      setShowDetails(temp);
+    }
+    handleDevices();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
   return (
     <StyledList>
-      {devices.map((device) => (
+      {showDetails.map((device) => (
         <StyledCard key={device.id}>
           <section>
             {CATEGORY_MAP[device.device_category]}
             <p>Device:{device.device}</p>
-            <p>Device category: {device.device_category}</p>
-            <p>Model: {device.model}</p>
-            <p>Power consumption {device.power_consumption} W/h</p>
-            <p>
-              Power consumption Standby: {device.power_consumption_standby} W/h
-            </p>
+            {device.isActive ? (
+              ""
+            ) : (
+              <>
+                <p>Location: {device.location}</p>
+                <p>Device category: {device.device_category}</p>
+                <p>Model: {device.model}</p>
+                <p>Power consumption {device.power_consumption} W/h</p>
+                <p>
+                  Power consumption Standby: {device.power_consumption_standby}{" "}
+                  W/h
+                </p>
+              </>
+            )}
             <p>Average usage time:{device.average_usage_time}h</p>
-            <p>Location: {device.location}</p>
           </section>
+          <button
+            onClick={() => {
+              handleButtonClick(device.id);
+            }}
+          >
+            Details
+          </button>
         </StyledCard>
       ))}
     </StyledList>
   );
 }
+
 const StyledCard = styled.li`
   border: solid black 3px;
 `;
```

Comment:
Your toggle functionality to display details is quite complex and I would suggest to use a simpler approach. My reasons:
- Your StyledCard is a reusable component and can be built as a standalone component. I would therefore also outsource this part to a separate component.
- Your toggle functionality is a feature of your Card component and can therefore be packed into it.
- This way you can drastically reduce your complicated logic that you packed into a useEffect and work with a toggleState inside your Card component. I have coded this as an example:

**pages/index.js**

```js
export default function Home({ devices }) {
  return (
    <StyledList>
      {devices.map((device) => (
        <Card
          key={device.id}
          deviceCategory={device.device_category}
          name={device.device}
          location={device.location}
          model={device.model}
          powerConsumption={device.power_consumption}
          powerConsupmtionStandby={device.power_consumption_standby}
          averageUsageTime={device.average_usage_time}
        />
      ))}
    </StyledList>
  );
}
```

**components/Card.js**

```js
import { useState } from "react";
import styled from "styled-components";
import { EntertainmentIcon } from "@/components/Icons";
import { AppliancesIcon } from "@/components/Icons";
import { WorkIcon } from "@/components/Icons";

const CATEGORY_MAP = {
  Entertainment: <EntertainmentIcon />,
  Appliances: <AppliancesIcon />,
  Work: <WorkIcon />,
};

const Wrapper = styled.li`
  border: solid black 3px;
`;

export default function Card({
  deviceCategory,
  name,
  location,
  model,
  powerConsumption,
  powerConsumptionStandby,
  averageUsageTime,
}) {
  const [areDetailsDisplayed, setAreDetailsDisplayed] = useState(false);

  return (
    <Wrapper>
      <section>
        {CATEGORY_MAP[deviceCategory]}
        <p>Device:{name}</p>
        {areDetailsDisplayed && (
          <ul>
            <li>Location: {location}</li>
            <li>Device category: {deviceCategory}</li>
            <li>Model: {model}</li>
            <li>Power consumption {powerConsumption} W/h</li>
            <li>Power consumption Standby: {powerConsumptionStandby} W/h</li>
          </ul>
        )}
        <p>Average usage time:{averageUsageTime}h</p>
      </section>
      <button onClick={() => setAreDetailsDisplayed(!areDetailsDisplayed)}>
        Details
      </button>
    </Wrapper>
  );
}
```

I hope my example code does help you in understanding my approach. I think this would make your life much more easier as you should try to avoid to use `useEffect` if it is not really necessary. :)